### PR TITLE
DP-426 Change button - Connected Persons - Load from DB, Update and Save

### PR DESCRIPTION
**Pull request by rmohammed-goaco**
https://noticingsystem.atlassian.net/browse/DP-426

### CODE REVIEW

File: Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    [Fact]
+    public async Task Execute_ThrowsUnknownOrganisationException_WhenOrganisationDoesNotExist()
+    {
+        var command = (organisationId: Guid.NewGuid(), connectedEntityId: Guid.NewGuid(), updateConnectedEntity: new UpdateConnectedEntity
+        {
+            EntityType = (ConnectedEntityType)0
+        });
+        _mockOrganisationRepository.Setup(repo => repo.Find(command.organisationId)).ReturnsAsync((Persistence.Organisation)null);
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs:38]: Static analysis reports a problem here: `Converting null literal or possible null value to non-nullable type.`.

File: Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        var updateConnectedEntity = new UpdateConnectedEntity
+        {
+            EntityType = (ConnectedEntityType)0
+        };
+        var command = (organisationId, connectedEntityId, updateConnectedEntity);
+
+        var organisation = FakeOrganisation();
+
+        _mockOrganisationRepository.Setup(repo => repo.Find(organisationId)).ReturnsAsync(organisation);
+        _mockConnectedEntityRepository.Setup(repo => repo.Find(organisationId, connectedEntityId)).ReturnsAsync((Persistence.ConnectedEntity)null);
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs:96]: Static analysis reports a problem here: Converting null literal or possible null value to non-nullable type..

File: Services/CO.CDP.Organisation.WebApi/Model/Command.cs [hunk 3]
```diff
@@ -181,6 +181,25 @@ public record RegisterConnectedEntity
     public DateTimeOffset? EndDate { get; init; }
 }
 
+public record UpdateConnectedEntity
+{
+    public string Id { get; init; }
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/Model/Command.cs:186]: Static analysis reports a problem here: `Non-nullable property 'Id' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.`.

File: Services/CO.CDP.Organisation.WebApi/UseCase/UpdateConnectedEntityUseCase.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+using CO.CDP.OrganisationInformation.Persistence;
+using ConnectedEntity = CO.CDP.OrganisationInformation.Persistence.ConnectedEntity;
+
+namespace CO.CDP.Organisation.WebApi.UseCase;
+
+public class UpdateConnectedEntityUseCase(
+    IConnectedEntityRepository connectedEntityRepository,
+    IOrganisationRepository organisationRepository,
+    IMapper mapper,
+    Func<Guid> guidFactory)
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/UseCase/UpdateConnectedEntityUseCase.cs:12]: Unused parameter.

**Review by jakzal — CHANGES_REQUESTED** [review]
Looks great, except for a few static analysis problems reported.

File: Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs (hunk 1 above)
Comment (rmohammed-goaco) [reply Services/CO.CDP.Organisation.WebApi.Tests/UseCase/UpdateConnectedEntityUseCaseTest.cs]: Thanks @jakzal . Have pushed an update to address this.

File: Frontend/CO.CDP.OrganisationApp/Pages/Supplier/ConnectedEntity/ConnectedEntityCheckAnswersOrganisation.cshtml.cs [hunk 5]
```diff
@@ -154,6 +192,139 @@ private void InitModal(ConnectedEntityState state)
...
+        }
+
+        var updateConnectedEntity = new UpdateConnectedEntity
+        (
+            id: ConnectedEntityId.ToString(),
+            addresses: addresses,
+            companyHouseNumber: state.CompaniesHouseNumber,
+            endDate: null,
+            entityType: state.ConnectedEntityType!.Value.AsApiClientConnectedEntityType(),
+            hasCompnayHouseNumber: state.HasCompaniesHouseNumber!.Value,
```
Comment (dpatel017) [inline Frontend/CO.CDP.OrganisationApp/Pages/Supplier/ConnectedEntity/ConnectedEntityCheckAnswersOrganisation.cshtml.cs:249]: This needs to map from state.SupplierHasCompanyHouseNumber

File: Frontend/CO.CDP.OrganisationApp/Pages/Supplier/ConnectedEntity/ConnectedEntityCheckAnswersOrganisation.cshtml.cs [hunk 6]
```diff
@@ -154,6 +192,139 @@ private void InitModal(ConnectedEntityState state)
...
+                category: state.ConnectedEntityOrganisationCategoryType!.Value.AsApiClientConnectedEntityOrganisationCategoryType(),
+                controlCondition: state.ControlConditions.AsApiClientControlConditionList(),
+                insolvencyDate: state.InsolvencyDate,
+                lawRegistered: state.LawRegistered,
+                name: state.OrganisationName,
+                organisationId: null,
+                registeredLegalForm: state.LegalForm
+            );
+        }
+        else
```
Comment (dpatel017) [inline Frontend/CO.CDP.OrganisationApp/Pages/Supplier/ConnectedEntity/ConnectedEntityCheckAnswersOrganisation.cshtml.cs:213]: we can remove this block from here as this page only for Organisation check answer

**Review by dpatel017 — APPROVED** [review]

**Review by dharmverma — APPROVED** [review]

**Review by jakzal — APPROVED** [review]

**Pull request merged**